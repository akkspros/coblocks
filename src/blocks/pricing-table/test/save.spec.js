/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render 2 columns', () => {
		block.attributes.count = 2;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-2-columns' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render 3 columns', () => {
		block.attributes.count = 3;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-3-columns' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
