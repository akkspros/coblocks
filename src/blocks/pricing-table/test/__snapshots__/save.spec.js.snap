// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/pricing-table should render 1`] = `
"<!-- wp:coblocks/pricing-table -->
<div class=\\"wp-block-coblocks-pricing-table has-2-columns has-text-align-center\\"><div class=\\"wp-block-coblocks-pricing-table__inner\\"></div></div>
<!-- /wp:coblocks/pricing-table -->"
`;

exports[`coblocks/pricing-table should render 2 columns 1`] = `
"<!-- wp:coblocks/pricing-table -->
<div class=\\"wp-block-coblocks-pricing-table has-2-columns has-text-align-center\\"><div class=\\"wp-block-coblocks-pricing-table__inner\\"></div></div>
<!-- /wp:coblocks/pricing-table -->"
`;

exports[`coblocks/pricing-table should render 3 columns 1`] = `
"<!-- wp:coblocks/pricing-table {\\"count\\":3} -->
<div class=\\"wp-block-coblocks-pricing-table has-3-columns has-text-align-center\\"><div class=\\"wp-block-coblocks-pricing-table__inner\\"></div></div>
<!-- /wp:coblocks/pricing-table -->"
`;

exports[`coblocks/pricing-table should render with content 1`] = `
"<!-- wp:coblocks/pricing-table {\\"count\\":3} -->
<div class=\\"wp-block-coblocks-pricing-table has-3-columns has-text-align-center\\"><div class=\\"wp-block-coblocks-pricing-table__inner\\"></div></div>
<!-- /wp:coblocks/pricing-table -->"
`;
